dj_rest_auth/views.py

```diff
@@ -107,11 +104,13 @@ def get_response(self):
                 instance=data,
                 context=self.get_serializer_context(),
             )
-        else:
+        elif self.token:
             serializer = serializer_class(
                 instance=self.token,
                 context=self.get_serializer_context(),
             )
+        else:
+            return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
We don't need to return a payload. If we are using session login, the response will contain a session cookie header.